main.go

```diff
@@ -20,7 +22,53 @@ func genLog() gol.Logger {
 	return gol.NewCustomLogger(dir + "/.git")
 }
 
+func changeToRepositoryRoot() error {
+	var errRootReached = errors.New("root reached")
```

Comment:
Interesting manner to manage error handling.

Reply:
Yeah since the error interface only provides the `.Error()` string method, if you want to handle a certain error in a specific way rather than bubble it up with the `if err != nil { return err }` pattern, you either have to

* define your error variables of interest and check for them using equality checks, or
* you could define your concrete error type that might have a `.Kind` property that encodes what error it is (but then you need type assertions to get at the concrete type and account for the case where the error is not of your type), or,
* look into the error string, which I'm not a fan of...

at least that's in my experience so far!

Reply:
Remember me custom exception I love create in PHP.